django/utils/url_config.py

```diff
@@ -0,0 +1,65 @@
+import inspect
+from collections import defaultdict
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import module_loading
+
+BACKENDS = defaultdict(dict)
+
+
+def add_backend(backend_type, scheme, path):
+    if scheme in BACKENDS[backend_type]:
+        raise ImproperlyConfigured('Scheme {0} has already been registered as a {1} handler'.format(scheme,
+                                                                                                    backend_type))
+    BACKENDS[backend_type][scheme] = path
```

Comment:
If a user who writes his own backend makes a typo for `backend_type`, this function will return without error but the registration will mysteriously not have worked. The two options I see are to make a whitelist of allowed `backend_type`s, or make this function private (by adding an underscore in front of the name) and just expose the convenience functions such as `register_db_backend`.

Reply:
Good point. I've made them private.